A43-sdk-authorization.md

```diff
@@ -0,0 +1,713 @@
+  						  "against, for potentially advanced use cases. The headers are"
+  						  "ANDed together, i.e., it matches only if *all* the headers"
+  						  "match.",
+  						"type": "array",
+  						"items": {
+  							"type": "object",
+  							"properties": {
+  								"key": {
+  									"description": "The name of the HTTP header to"
+  									  "match. The following headers are *not*"
+  									  "supported: the \"Host\" header, \"hop-by-hop\""
+  									  "headers (e.g. those listed in \"Connection\""
+  									  "header), HTTP/2 pseudo headers (\":\"-prefixed)"
+  									  "and headers prefixed with \"grpc-\".",
+  									"type": "string"
+  								},
+  								"values": {
+  									"description": "A list of header values to"
+  									  "match. The header values are ORed together,"
+  									  "i.e., it matches if one of the values"
+  									  "matches. Multi-valued headers are considered"
+  									  "a single value with commas added between"
+  									  "values."
+  									  "This field supports Exact, Prefix, Suffix"
+  									  "and Presence match."
+  									  "- Exact match: \"abc\" will match on value"
+  									  "  \"abc\"."
+  									  "- Prefix match: \"abc*\" will match on value"
+  									  "  \"abc\" and \"abcd\"."
+  									  "- Suffix match: \"*abc\" will match on value"
+  									  "  \"abc\" and \"xabc\"."
+  									  "- Presence match: \"*\" will match when the"
+  									  "  value is not empty.",
+  									"type": "array",
+  									"items": {
+  										"type": "string"
+  									}
+  								}
+  							},
+  							"required": ["key", "values"]
+  						}
+  					}
+  				}
+  			}
+  		},
+  		"required": ["name"]
+  	}
+  },
+  "description": "AuthorizationPolicy defines which principals are permitted to"
+    "access which resource. Resources are RPC methods scoped by services.",
+  "type": "object",
+  "properties": {
+  	"name": {
+  		"description": "The name of an authorization policy. It is mainly for"
+  		  "monitoring and error message generation.",
+  		"type": "string"
+  	},
+  	"deny_rules": {
+  		"description": "List of deny rules to match. If a request matches any of the"
+  		  "deny rules, then it will be denied. If none of the deny rules matches or"
+  		  "there are no deny rules, the allow rules will be evaluated.",
+  		"type": "array",
+  		"items": {
+  			"$ref": "#/definitions/rule"
+  		}
+  	},
+  	"allow_rules": {
+  		"description": "List of allow rules to match. The allow rules will only be"
+  		  "evaluated after the deny rules. If a request matches any of the allow"
+  		  "rules, then it will allowed. If none of the allow rules matches, it will"
+  		  "be denied.",
+  		"type": "array",
+  		"items": {
+  			"$ref": "#/definitions/rule"
+  		}
+  	}
+  },
+  "required": ["name", "allow_rules"]
+}
+```
+
+#### Details
+
+gRPC SDK authorization policy has a list of *allow rules* and a list of *deny rules*.
```

Comment:
The description here assumes a single authz policy attached to a server. Even if your JSON schema doesn't allow multiple policies what happens if you attach multiple policies like this (in Java):

```
authzServerInterceptor1 = FileAuthorizationServerInterceptor.create(authzPolicyFile1);
authzServerInterceptor2 = FileAuthorizationServerInterceptor.create(authzPolicyFile2);

Server server =
    Grpc.newServerBuilderForPort(port, serverCreds)
        .addService(service)
        .intercept(authzServerInterceptor1)
        .intercept(authzServerInterceptor2)
        .build()
        .start();
```

Is this allowed or will this be reported as an error? If not an error how are multiple policies handled?